### DISCUSSION TITLE
Empty dialog which can't be closed

### DISCUSSION THREAD

**Discussion by beatrixwillius** [phcode-dev/phoenix#1851] [Q&A] [answered]
When Phoenix started I blocked some websites with Little Snitch which looked dicey. As I wanted to quit I got the nice empty dialog which can't be closed by clicking on the close button:

![Screenshot 2024-09-16 at 08 17 16](https://github.com/user-attachments/assets/b49fc5b8-864c-4925-9732-5a4cee5862f0)

Little Snitch:

![Screenshot 2024-09-16 at 08 19 46](https://github.com/user-attachments/assets/c1db8cd9-0346-435b-8b54-ce432e058477)

Why is Google Tag Manager needed at all?

**Comment by abose** [comment, marked as answer]
The empty dialog is a survey that we show at https://s.surveyplanet.com/jssqbld8 which I see is blocked. It can usually be dismissed and not a critical component, but i'm unsure why it bricked the system. have to test the scenario.

We use google tag manager for analytics, but be assured that we have done everything that it takes to anonymize you from google contexts. We load ga in a sandboxed window in desktop apps for anonymizing your profile and preventing google from mapping you to your google account. The window that loads ga is isolated into its own physical browser window context, and does not live in the same window as the code editor, so its impossible that ga have access to what you are doing in phoenix. This is the code that [separates](https://github.com/phcode-dev/phoenix-desktop/blob/main/src-tauri/tauri.conf.json#L473) the window and this is the [code that loads ga](https://github.com/phcode-dev/phoenix/blob/main/src/desktop-metrics.html). Blocking ga should theoretically not brick the system as its a non essential component. 

We are also privacy conscious as you are, so we went though all the extra lengths to address your concern as its the same we had too. We primarily develop Phoenix on Linux for this reason, as we don't trust Apple either as much as we don't trust Google or Microsoft with our data. We went so far as to create our own metrics service for this exact reason: https://github.com/aicore/core-analytics-client-lib . But we didn't go that route as we figured out a way to sandbox and anonymize ga, which is the only free alternative out there at our scale.

 We have specified it in our privacy notice about the same.  It can be fully disabled by going to `help- health report`. 
<img width="253" alt="image" src="https://github.com/user-attachments/assets/7a1ff332-ad56-45ed-a73f-a3b685562e97"> 

Opening phoenix code again and disabling health reports should fix the above issue for your use case though. Did it answer your queries?
